cesium/writer_stream.go

```diff
@@ -420,7 +428,7 @@ func (w *idxWriter) Write(fr Frame) (Frame, error) {
 	return fr, nil
 }
 
-func (w *idxWriter) Commit(ctx context.Context) (telem.TimeStamp, error) {
+func (w *idxWriter) Commit(ctx context.Context, autoPersist bool) (telem.TimeStamp, error) {
```

Comment:
Why do we need to pass `autoPersist` around everywhere? Can't we pass this as part of the configuration?

Reply:
If you need to do this because we have a `writer.SetMode` function and we need to do this dynamically. If so, let's get rid of the ability to `SetMode` on the writer and just make it so you need to re-instantiate the writer if you want to turn auto commit on or off.